public-post-preview.php

```diff
@@ -649,7 +649,7 @@ private static function verify_nonce( $nonce, $action = -1 ) {
 	 * @return array The post ids. (Empty array if no ids are registered.)
 	 */
 	private static function get_preview_post_ids() {
-		return get_option( 'public_post_preview', array() );
+		return array_map( 'intval', get_option( 'public_post_preview', array() ) );
```

Comment:
Added intval so strict check in_array doesn't need to convert array to int each time.

Reply:
Most data being saved is giving post id as int except 1 from `post_updated` hook.
I've updated `unregister_public_preview_on_edit` but we also need to take care of existing serialized data in DB.